diff --git a/configure b/configure
index ca7bd8d..19fdd3e 100755
--- a/configure
+++ b/configure
@@ -49650,7 +49650,7 @@ then :
   printf %s "(cached) " >&6
 else case e in #(
   e) r_save_LIBS="${LIBS}"
-LIBS="${LIBS} -licuuc -licui18n"
+LIBS="${LIBS} -licui18n -licuuc -licudata -lpthread -lm /usr/lib/libstdc++.a"
 if test "$cross_compiling" = yes
 then :
   r_cv_icu=no
@@ -49702,7 +49702,7 @@ if test "x${r_cv_icu}" = xyes; then
 
 printf "%s\n" "#define USE_ICU 1" >>confdefs.h
 
-  LIBS="${LIBS} -licuuc -licui18n"
+  LIBS="${LIBS} -licui18n -licuuc -licudata -lpthread -lm /usr/lib/libstdc++.a"
 else
   use_ICU=no
 fi
diff --git a/src/main/Makefile.in b/src/main/Makefile.in
index a4180e8..7386853 100644
--- a/src/main/Makefile.in
+++ b/src/main/Makefile.in
@@ -37,7 +37,7 @@ SOURCES_C = \
 	times.c \
 	unique.c util.c \
 	version.c \
-	g_alab_her.c g_cntrlify.c g_fontdb.c g_her_glyph.c
+	g_alab_her.c g_cntrlify.c g_fontdb.c g_her_glyph.c Lapack.c
 
 SOURCES_F =  xxxpr.f
 
@@ -147,7 +147,7 @@ R: Makedeps
 	(cd $(@D); $(MAKE) $(@F))
 
 $(R_binary): $(R_bin_OBJECTS) $(R_bin_DEPENDENCIES)
-	$(MAIN_LINK) -o $@ $(R_bin_OBJECTS) $(R_bin_LDADD)
+	$(MAIN_LINK) -o $@ $(R_bin_OBJECTS) $(R_bin_LDADD) -lreadline -static-libgcc
 
 libR.a: $(OBJECTS) $(STATIC_LIBS)
 	-@mkdir libs
diff --git a/src/modules/lapack/Makefile.in b/src/modules/lapack/Makefile.in
index 0a1c9cb..4689808 100644
--- a/src/modules/lapack/Makefile.in
+++ b/src/modules/lapack/Makefile.in
@@ -100,7 +100,7 @@ R: Makedeps
 	@$(MAKE) $(lapack_la)
 
 $(lapack_la): $(lapack_la_OBJECTS)
-	$(SHLIB_LINK) -o $@ $(lapack_la_LDFLAGS) $(lapack_la_OBJECTS) $(lapack_la_LIBADD) $(EXTRA_LIBS)
+	$(SHLIB_LINK) -o $@ $(lapack_la_LDFLAGS) $(lapack_la_OBJECTS) $(lapack_la_LIBADD) $(EXTRA_LIBS) -static-libgcc
 ## <FIXME>
 ## Not sure if we want to do this ...
 	@$(MAKE) rhome="$(abs_top_builddir)" install
