--- R-4.4.2-old/configure	2024-10-31 08:01:28.000000000 +0000
+++ R-4.4.2/configure	2024-11-27 06:38:21.520000365 +0000
@@ -49650,7 +49650,7 @@
   printf %s "(cached) " >&6
 else case e in #(
   e) r_save_LIBS="${LIBS}"
-LIBS="${LIBS} -licuuc -licui18n"
+LIBS="${LIBS} -licui18n -licuuc -licudata /usr/lib/gcc/aarch64-linux-gnu/7/libstdc++.a -laf -lpthread -lm -ldl"
 if test "$cross_compiling" = yes
 then :
   r_cv_icu=no
@@ -49702,7 +49702,7 @@
 
 printf "%s\n" "#define USE_ICU 1" >>confdefs.h
 
-  LIBS="${LIBS} -licuuc -licui18n"
+  LIBS="${LIBS} -licui18n -licuuc -licudata -lpthread /usr/lib/gcc/aarch64-linux-gnu/7/libstdc++.a -laf -lpthread -lm -ldl"
 else
   use_ICU=no
 fi
